File: java/client/test/com/thoughtworks/selenium/corebased/TestTypeRichText.java
package com.thoughtworks.selenium.corebased;

import com.thoughtworks.selenium.InternalSelenseTestBase;

import org.junit.After;
import org.junit.Test;

public class TestTypeRichText extends InternalSelenseTestBase {
  @After
  public void resetFrame() {
    selenium.selectFrame("relative=top");
  }

  @Test
  public void testTypeRichText() throws Exception {
    String isIe = selenium.getEval("browserVersion.isIE");
    if (Boolean.valueOf(isIe)) {
      return;
    }

    selenium.open("../tests/html/test_rich_text.html");
    selenium.selectFrame("richtext");
    verifyEquals(selenium.getText("//body"), "");
    selenium.type("//body", "hello world");
    verifyEquals(selenium.getText("//body"), "hello world");
  }
}

File: dotnet/test/Selenium.WebDriverBackedSelenium.Tests/TestTypeRichText.cs
using System;
using System.Collections.Generic;
using System.Text;
using NUnit.Framework;
using System.Text.RegularExpressions;
using System.Threading;

namespace Selenium.Tests
{
    [TestFixture]
    public class TestTypeRichText : SeleniumTestCaseBase
    {
        [Ignore("GetEval of browserVersion fails.")]
        public void ShouldBeAbleToTypeInRichText()
        {
            String isIe = selenium.GetEval("browserVersion.isIE");
            if (bool.Parse(isIe))
            {
                return;
            }

            selenium.Open("../tests/html/test_rich_text.html");
            selenium.SelectFrame("richtext");
            Assert.AreEqual(selenium.GetText("//body"), "");
            selenium.Type("//body", "hello world");
            Assert.AreEqual(selenium.GetText("//body"), "hello world");
        }

        [TearDown]
        public void resetFrame()
        {
            selenium.SelectFrame("relative=top");
        }
    }
}
